moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/detail/ompl_constraints.h

```diff
@@ -0,0 +1,293 @@
+  robot_model::RobotModelConstPtr robot_model_;
+  robot_state::RobotStatePtr robot_state_;
+  const robot_state::JointModelGroup* joint_model_group_;
+
+  /** \brief Robot link the constraints are applied to. */
+  std::string link_name_;
+
+  /** \brief Upper and lower bounds on constrained variables. */
+  std::vector<Bounds> bounds_;
+
+  /** \brief target for equality constraints, nominal value for inequality constraints. */
+  Eigen::Vector3d target_position_;
+
+  /** \brief target for equality constraints, nominal value for inequality constraints. */
+  Eigen::Quaterniond target_orientation_;
+
+public:
+  // Macro for classes containing fixed size eigen vectors that are dynamically allocated when used.
+  // https://eigen.tuxfamily.org/dox/group__TopicStructHavingEigenMembers.html
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+};
+
+/** \brief Box shaped position constraints
+ *
+ * Reads bounds on x, y and z position from a position constraint
+ * at constraint_region.primitives[0].dimensions.
+ * Where the primitive has to be of type BOX.
+ *
+ * These bounds are applied around the nominal position and orientation
+ * of the box.
+ *
+ * */
+class PositionConstraint : public BaseConstraint
+{
+public:
+  PositionConstraint(robot_model::RobotModelConstPtr robot_model, const std::string& group, const unsigned int num_dofs)
+    : BaseConstraint(robot_model, group, num_dofs)
+  {
+  }
+  virtual void parseConstraintMsg(const moveit_msgs::Constraints& constraints) override;
+  virtual Eigen::VectorXd calcError(const Eigen::Ref<const Eigen::VectorXd>& x) const override;
+  virtual Eigen::MatrixXd calcErrorJacobian(const Eigen::Ref<const Eigen::VectorXd>& x) const override;
+};
+
+/** \brief orientation constraints based on angle-axis error.
+ *
+ * (aka exponential coordinates)
+ * This is analog to how orientation Error is calculated in the TrajOpt motion planner.
+ * It is NOT how orientation error is handled in the default MoveIt constraint samplers.
+ * (There, XYZ intrinsic euler angles are used.)
+ *
+ * */
+class AngleAxisConstraint : public BaseConstraint
+{
+public:
+  AngleAxisConstraint(robot_model::RobotModelConstPtr robot_model, const std::string& group,
+                      const unsigned int num_dofs)
+    : BaseConstraint(robot_model, group, num_dofs)
+  {
+  }
+
+  virtual void parseConstraintMsg(const moveit_msgs::Constraints& constraints) override;
+  virtual Eigen::VectorXd calcError(const Eigen::Ref<const Eigen::VectorXd>& x) const override;
+  virtual Eigen::MatrixXd calcErrorJacobian(const Eigen::Ref<const Eigen::VectorXd>& x) const override;
+};
+
+/** \brief Position and orientation constraint
+ *
+ * Combine position constraints modelled as a box with orientation constraints
+ * using exponential coordinates, but ignore the last rotation around the z-axis.
+ * This is a common constraint for grasping, welding, ...
+ * */
+class PoseConstraint : public BaseConstraint
```

Comment:
I will change this base class to handle simple equality constraints for the first version. Then I can add this current base class as a new derived class that handles bounds, later on, something like `BoundsConstraint` or something.